assets/aml-benchmark/scripts/custom_inference_postprocessors/humaneval_postprocessor_template.py

```diff
@@ -0,0 +1,258 @@
+    if prediction_dataset:
+        pred_data = _read_jsonl_file(prediction_dataset)
+        if ground_truth_dataset:
+            task_id = _read_jsonl_file(ground_truth_dataset)
+            pred_with_task_id = merge_id(task_id, pred_data)
+
+    regex_expr = REGEX_EXPR
+
+    # Post processing the prediction and ground truth columns
+    ground_truths, predictions = run_humaneval_postprocessor(pred_with_task_id, regex_expr)
+
+    _write_to_jsonl_file(predictions, output_path, ground_truths)
+
+
+def merge_id(
+    label_data: List[Dict[str, Any]],
+    pred_data: List[Dict[str, Any]]
+) -> Union[pd.DataFrame, List[Dict[str, Any]]]:
+    """
+    Run the custom processor function to extract the ground truth.
+
+    :param data: Data loaded from _read_jsonl_file function.
+    :type: List[Dict[str, Any]]
+    :return: pd.DataFrame or List[Dict[str, Any]]]
+    """
+    # Get the original data from Hugging Face
+    expected_op = [{**x, **y} for x, y in zip(label_data, pred_data)]
+    return expected_op
+
+
+def run_humaneval_postprocessor(
+    data: List[Dict[str, Any]],
+    regex_exp: str = None
+) -> Union[pd.DataFrame, List[Dict[str, Any]]]:
+    """
+    Run the custom processor function to extract the ground truth.
+
+    :param data: Data loaded from _read_jsonl_file function.
+    :type: List[Dict[str, Any]]
+    :param regex_exp: Regex expression to extract the prediction.
+    :type: str
+    :return: pd.DataFrame or List[Dict[str, Any]]]
+    """
+    gt_list = []
+    pred_list = []
+
+    # Get the original data from Hugging Face
+    original_dataset = load_dataset("openai_humaneval", split="test").to_pandas()
+    pred_df = pd.DataFrame(data)
+
+    # Merge the prediction dataset with the original dataset
+    pred_df_full = pd.merge(pred_df,
+                            original_dataset,
+                            how='left',
+                            left_on='ground_truth',
+                            right_on='task_id')
+
+    # Rename the prediction column
+    pred_df_full.rename(columns={"prediction": "original_prediction"},
+                        inplace=True)
+
+    # Convert the dataframe to a dictionary of lists of each row
+    pred_dict_full = pred_df_full.to_dict('records')
+
+    # Post processing the prediction and ground truth columns
+    for idx, row in enumerate(pred_dict_full):
+        gt = "\n" + row["test"] + "\n" + "check(" + row["entry_point"] + ")"
+
+        if str("def " + row["entry_point"] + "(") in row["original_prediction"]:
+            # If the model regenerates the prompt/ function name
+            pred_combined_prompt = row["original_prediction"]
+        else:
+            pred_combined_prompt = row["prompt"]+"\n"+row["original_prediction"]
+
+        # Applying regex on the prediction column
+        if regex_exp:
+            pred = apply_regex_expr(pred_combined_prompt, regex_exp)
+        else:
+            pred = pred_combined_prompt
+        failed, _ = code_run(pred,
```

Comment:
I am assuming all of this code to run the output will be cleaned up from the post processor?

Reply:
Yes, the_code_run() would be removed. The compute metrics component should show whether a prediction passed or failed the test cases. This is not available in compute metrics (atleast the component i used.) This is quite important for debugging IMO.

Reply:
@EricWrightAtWork suggested we keep these. It would be helpful in debugging unless we have this capability in compute_metrics,